`test: check if binary has executable stack` fails on HPPA (arch always needs exec. stacks)

**Describe the bug**

zstd's test suite fails on HPPA. Specifically, the `test: check if binary has executable stack` test fails because of an executable stack. On HPPA, this is actually expected, and the test should be skipped/XFAILed.

**To Reproduce**
Steps to reproduce the behavior:
1. Run zstd-1.5.5's test suite on HPPA.
2. Observe failure:
```
$ make check
[...]
4+0 records in
4+0 records out
4 bytes copied, 0 s, Infinity B/s
tmp_corrupt.zst     : 65537 bytes
tmp_corrupt.zst     : 65537 bytes
tmp.zst             : 65537 bytes
test: check if binary has executable stack (#2963)
zstd binary has executable stack!
make[1]: *** [Makefile:343: test-zstd] Error 1
make[1]: Leaving directory '/tmp/zstd/tests'
make: *** [Makefile:91: shortest] Error 2
```

All other tests pass if I comment out this test.

**Expected behavior**

All tests to pass and for the execstack test to be skipped where it's not relevant.

**Desktop (please complete the following information):**
 - OS: Gentoo Linux
 - Version: 1.5.5, master
 - Compiler: gcc
 - Flags: O2
 - Other relevant hardware specs: HPPA/PARISC architecture
 - Build system: Makefile, meson

**Additional context**

HPPA requires executable stacks right now during normal operations, so generating an executable stack here is no cause for concern.

Originally reported downstream in Gentoo at https://bugs.gentoo.org/903923.

--- a/build/meson/meson.build
+++ b/build/meson/meson.build
@@ -116,10 +116,16 @@ if [compiler_gcc, compiler_clang].contains(cc_id)
   if cc_id == compiler_clang
     common_warning_flags += ['-Wconversion', '-Wno-sign-conversion', '-Wdocumentation']
   endif
-  cc_compile_flags = cc.get_supported_arguments(common_warning_flags + ['-Wstrict-prototypes'])
-  cxx_compile_flags = cxx.get_supported_arguments(common_warning_flags)
+  noexecstack_flags = ['-Wa,--noexecstack' ]
+  noexecstack_link_flags = ['-Wl,-z,noexecstack']
+  cc_compile_flags = cc.get_supported_arguments(common_warning_flags + noexecstack_flags + ['-Wstrict-prototypes'])
+  cxx_compile_flags = cxx.get_supported_arguments(common_warning_flags + noexecstack_flags)
   add_project_arguments(cc_compile_flags, language : 'c')
   add_project_arguments(cxx_compile_flags, language : 'cpp')
+  cc_link_flags = cc.get_supported_link_arguments(noexecstack_link_flags)
+  cxx_link_flags = cxx.get_supported_link_arguments(noexecstack_link_flags)
+  add_project_link_arguments(cc_link_flags, language: 'c')
+  add_project_link_arguments(cxx_link_flags, language: 'cpp')
 elif cc_id == compiler_msvc
   msvc_compile_flags = [ '/D_UNICODE', '/DUNICODE' ]
   if use_multi_thread

